### PR TITLE
Components do not have to implement a base class anymore

### PR DESCRIPTION
**Pull request by mnapoli**
In this PR I prepare the work to allow components to not implement the base class.

The main change is that Compose can no longer access `component.id`, `component.context` or `component.inputs` (because we have no guarantee these properties are exposed by component classes).

Instead of forcing components to expose these properties, it was fairly simple to refactor the code to get "id", "context" and "inputs" from other variables.

Matching PR: https://github.com/serverless/components-core/pull/8

### CODE REVIEW

File: src/Component.js [hunk 1]
```diff
@@ -1,17 +1,6 @@
 'use strict';
 
 class Component {
```
Comment (mnapoli) [inline src/Component.js:3]: Note: this class will eventually be removed/replaced by `@serverless-components/core`.

Comment (pgrzesik) [reply src/Component.js:3]: Is this even needed then for anything? The Framework component doesn't use it at all

Comment (mnapoli) [reply src/Component.js:3]: Mmmh you're right 🤔 

Let me remove it actually

**Review by pgrzesik — APPROVED** [review]
👍

**Pull request merged**